Automattic/wp-calypso
# Stats: Some cleanup of PromoCards wrapper component

#### Proposed Changes

- Renames the `MobilePromoCardWrapper` component to `PromoCards`
- Renames the containing directory to match component name
- Cleans up the display logic and HTML output a bit based on feedback in #71201 

#### Testing Instructions



1. Visit the Calypso Live link.
2. Visit the Stats page.
3. Scroll down and confirm the `DotPager` card is present.
4. Confirm the Yoast SEO card and mobile promo cards display properly.

**Note:** See #71201 for more in-depth testing steps.

#### Pre-merge Checklist





Related to #71201
